angrmanagement/ui/view_manager.py

```diff
@@ -136,12 +135,20 @@ def get_center_views(self):
     def first_view_in_category(self, category) -> Optional['BaseView']:
         """
         Return the first view in a specific category.
+        If the category is disassembly, return the current
+        category view else return the first view in
+        disassembly.
```

Comment:
Why do we want to do this? The method name is `first_view_in_category`, which means we should _always_ return the first view in each category, not the current view..

Reply:
The reason I implemented this in the first_view_in_category function was because its used quite a bit, especially in generating the content on the disassembly view. It was a simple workaround to a larger issue I think. I've been working on trying to have a list constantly updated according to the current view history, but haven't been able to figure out how to detect when a user changes tabs (hence updating the view history list). 

I also tried to just copy the logic of this function and create a separate one and replace all instances of `self.first_view_in_category('disassembly)` but I haven't been able to determine what is going wrong.

Still working on this change.

Reply:
> The reason I implemented this in the first_view_in_category function was because its used quite a bit, especially in generating the content on the disassembly view. It was a simple workaround to a larger issue I think. 

What is the issue that you are trying to address?

Reply:
> > The reason I implemented this in the first_view_in_category function was because its used quite a bit, especially in generating the content on the disassembly view. It was a simple workaround to a larger issue I think.
> 
> What is the issue that you are trying to address?

There was a lot of code using the `first_view_in_category` function, but if we are starting to add multiple views per view type, this function will become obsolete for anything other than the initialization of the default views.

Reply:
I think the correct behavior of `first_view_in_category` is to return the _most recently accessed_ view of that category. That would fall in line with the way the function is used and the way similar functionality works in IDA. I do however agree that there are a lot of spurious uses of the function in general, in places where it would be a lot cleaner for an object to have an explicit reference to it parent view.

Reply:
> I do however agree that there are a lot of spurious uses of the function in general, in places where it would be a lot cleaner for an object to have an explicit reference to it parent view.

Let's not touch this function in this PR. We can make a separate PR for changing the behavior of this function.

Moreover, you should never change the docstring and the code of a function so that its behavior conflicts with what its name suggests.

Reply:
> > I do however agree that there are a lot of spurious uses of the function in general, in places where it would be a lot cleaner for an object to have an explicit reference to it parent view.
> 
> Let's not touch this function in this PR. We can make a separate PR for changing the behavior of this function.
> 
> Moreover, you should never change the docstring and the code of a function so that its behavior conflicts with what its name suggests.

I went ahead and created a separate `current_view_in_category` function, but I changed code around the calling function of `first_view_in_category`. Now, anytime there is _only_ the first view of the Disassembly view, it will call `first_view_in_category` as it did before. But as soon as a new view is added, the `current_view_in_category` function will be called instead.